### PR TITLE
[To Feature] Improve loader features on authenticated engagement search page

### PR DESCRIPTION
**Pull request by NatSquared**
Issue #: https://citz-gdx.atlassian.net/browse/DESENG-664

*Description of changes:*
Improved & optimized changes to the engagement search features for DESENG-664

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/engagement/listing/index.tsx [hunk 1]
```diff
@@ -30,88 +30,79 @@ import AdvancedSearch from './AdvancedSearch/SearchComponent';
...
+                sort_key: paginationOptions.nested_sort_key?.toString(),
+                sort_order: paginationOptions.sort_order as 'asc' | 'desc' | undefined,
+                search_text: searchFilter.value,
+                engagement_status: searchOptions.status_list.map((status) => status.toString()),
+                created_from_date: searchOptions.created_from_date || undefined,
+                created_to_date: searchOptions.created_to_date || undefined,
+                published_from_date: searchOptions.published_from_date || undefined,
+                published_to_date: searchOptions.published_to_date || undefined,
+            };
+            // Filter out properties with empty strings or undefined values
```
Comment (jareth-whitney) [inline met-web/src/components/engagement/listing/index.tsx:94]: This is a nice little piece of code! Are any of the search params required or are all of them optional?

File: met-web/src/components/engagement/listing/index.tsx [hunk 2]
```diff
@@ -406,13 +400,17 @@ const EngagementListing = () => {
...
                             title={''}
+                            inputProps={{ 'aria-label': 'Search by name' }}
                             id="engagement-name"
                             data-testid="engagement/listing/searchField"
                             placeholder="Search by name"
                             name="searchText"
                             value={searchText}
                             sx={{ height: '40px', pr: 0 }}
                             onChange={(value) => setSearchText(value)}
+                            onKeyDown={(e) => {
```
Comment (jareth-whitney) [inline met-web/src/components/engagement/listing/index.tsx:411]: Always important to capture that enter key!

File: met-web/src/components/engagement/listing/index.tsx [hunk 3]
```diff
@@ -30,88 +30,79 @@ import AdvancedSearch from './AdvancedSearch/SearchComponent';
...
-    const pageFromURL = searchParams.get('page');
-    const sizeFromURL = searchParams.get('size');
+    const [searchParams, setSearchParams] = useSearchParams();
     const [searchFilter, setSearchFilter] = useState<SearchFilter>({
         key: 'name',
         value: '',
     });
-    const [searchText, setSearchText] = useState('');
-    const { engagements } = useRouteLoaderData('engagement-listing') as { engagements: Engagement[] };
+    const fetcher = useFetcher();
```
Comment (jareth-whitney) [inline met-web/src/components/engagement/listing/index.tsx:49]: Great to see the fetcher implemented.

File: met-web/src/components/engagement/listing/index.tsx [hunk 4]
```diff
@@ -30,88 +30,79 @@ import AdvancedSearch from './AdvancedSearch/SearchComponent';
...
+import { Page } from 'services/type';
+import { GetEngagementsParams } from 'services/engagementService';
 
 interface SearchFilter {
     key?: string;
     value?: string;
 }
 
-// Create a search, filter, and pagination string to pass values to the route.
-const updateURL = (
```
Comment (jareth-whitney) [inline met-web/src/components/engagement/listing/index.tsx:40]: RIP, it will not be missed 😂

File: met-web/src/routes/AuthenticatedRoutes.tsx [hunk 5]
```diff
@@ -87,13 +87,15 @@ const AuthenticatedRoutes = () => {
                         <Route path="form" element={<EngagementForm />} />
                     </Route>
                     <Route path="view" element={<EngagementView />} />
-                    <Route path=":slug" element={<EngagementView />} />
+                    <Route path="comments/:dashboardType" element={<EngagementComments />} />
```
Comment (jareth-whitney) [inline met-web/src/routes/AuthenticatedRoutes.tsx:90]: Great to see the routing suggestion implemented here as well.

**Review by jareth-whitney — APPROVED** [review]
Thanks for taking this on and optimizing things further. The only question I have is whether every search param is optional.

File: met-web/src/components/engagement/listing/index.tsx (hunk 1 above)
Comment (NatSquared) [reply met-web/src/components/engagement/listing/index.tsx:103]: They're optional! Making a request with no parameters uses the defaults on the API side

**Pull request merged**